api.go

```diff
@@ -23,28 +23,21 @@ type API struct {
 
 // NewAPI Creates a new API object for keeping semi-global settings to WebRTC objects
 func NewAPI(options ...func(*API)) *API {
-	a := &API{interceptor: &interceptor.NoOp{}}
+	a := &API{
+		interceptor:         &interceptor.NoOp{},
+		settingEngine:       &SettingEngine{},
+		mediaEngine:         &MediaEngine{},
+		interceptorRegistry: &interceptor.Registry{},
+	}
 
 	for _, o := range options {
 		o(a)
 	}
 
-	if a.settingEngine == nil {
```

Comment:
Can you also add a `ni` check to `WithSettingEngine` to match WithMediaEngine + WithInterceptorRegistry.

Reply:
Done! But I think this nil check is not necessary because in WithSettingEngine function we pass SettingEngine parameter by value - not by pointer.